e2e-polybft/consensus_test.go

```diff
@@ -126,41 +133,97 @@ func TestE2E_Consensus_RegisterValidator(t *testing.T) {
+
+	// create the second account and extract the address
+	addrs, err = cluster.InitSecrets(secondValidatorDataDir, 1)
+	require.NoError(t, err)
 
-	// assert that account is created
+	secondValidatorAddr := ethgo.Address(addrs[0])
+
+	// assert that accounts are created
 	validatorSecrets, err := genesis.GetValidatorKeyFiles(cluster.Config.TmpDir, cluster.Config.ValidatorPrefix)
 	require.NoError(t, err)
-	require.Equal(t, validatorSize+1, len(validatorSecrets))
+	require.Equal(t, validatorSize+2, len(validatorSecrets))
+
+	// collect owners validator secrets
+	ownerSecrets := validatorSecrets[0]
 
 	// wait for consensus to start
 	require.NoError(t, cluster.WaitForBlock(1, 10*time.Second))
 
-	// register new validator
-	require.NoError(t, srv.RegisterValidator(newValidatorSecrets, newValidatorBalanceRaw, newValidatorStakeRaw))
+	// owner enlists both new validators
+	require.NoError(t, owner.EnlistValidator(firstValidatorAddr.String(), ownerSecrets))
+	require.NoError(t, owner.EnlistValidator(secondValidatorAddr.String(), ownerSecrets))
 
-	go func() {
-		// start new validator
-		cluster.InitTestServer(t, 6, true, false)
-	}()
+	// start the first and the second validator
+	cluster.InitTestServer(t, validatorSize+1, true, false)
+	cluster.InitTestServer(t, validatorSize+2, true, false)
+
+	ownerAcc, err := sidechain.GetAccountFromDir(path.Join(cluster.Config.TmpDir, ownerSecrets))
+
+	// get the initial balance of the new validator
+	initialBalance, ok := new(big.Int).SetString(newValidatorInitBalance, 10)
+	require.True(t, ok)
+
+	// send some tokens from the owner to the first validator so that the first validator can register and stake
+	receipt1, err := txRelayer.SendTransaction(&ethgo.Transaction{
+		From:  ownerAcc.Ecdsa.Address(),
+		To:    &firstValidatorAddr,
+		Value: initialBalance,
+	}, ownerAcc.Ecdsa)
+	require.NoError(t, err)
+	require.Equal(t, uint64(types.ReceiptSuccess), receipt1.Status)
+
+	// send some tokens from the owner to the second validator so that the second validator can register and stake
+	receipt2, err := txRelayer.SendTransaction(&ethgo.Transaction{
+		From:  ownerAcc.Ecdsa.Address(),
+		To:    &secondValidatorAddr,
+		Value: initialBalance,
+	}, ownerAcc.Ecdsa)
+	require.NoError(t, err)
+	require.Equal(t, uint64(types.ReceiptSuccess), receipt2.Status)
+
+	// collect the first and the second validator from the cluster
+	firstValidator := cluster.Servers[validatorSize]
+	secondValidator := cluster.Servers[validatorSize+1]
+
+	// wait for the first validator's balance to be received
+	firstBalance, err := waitForNonZeroBalance(firstValidator, firstValidatorAddr, 5*time.Second)
+	require.NoError(t, err)
+	t.Logf("First validator balance=%d\n", firstBalance)
+
+	// wait for the first validator's balance to be received
+	secondBalance, err := waitForNonZeroBalance(secondValidator, secondValidatorAddr, 5*time.Second)
+	require.NoError(t, err)
+	t.Logf("Second validator balance=%d\n", secondBalance)
```

Comment:
Why do you need this check? Isn't it enough that you use tx relayer and waiting for receipt is blocking operation. This means that until both balances are increased, this part of the code should not be triggered. Am I missing out something?